src/script/sign.cpp

```diff
@@ -570,14 +571,17 @@ class DummySignatureCreator final : public BaseSignatureCreator {
     bool CreateSchnorrSig(const SigningProvider& provider, std::vector<unsigned char>& sig, const XOnlyPubKey& pubkey, const uint256* leaf_hash, const uint256* tweak, SigVersion sigversion) const override
     {
         sig.assign(64, '\000');
+        if (m_include_sighash) {
+            sig.push_back((unsigned char)SIGHASH_ALL);
```

Comment:
Could just do `sig.assign(64 + m_include_sighash, '\000');` in the previous line.

Reply:
IMO that's a bit harder to read and I don't like the implicit bool to int conversion.

Reply:
Have to agree with @achow101 here. Even though the alternative formulation is shorter ,and correct, I understand the intent of the code better this way.